# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/PBackground.ipdl
# Commit: bb63792c3333
# Full Hash: bb63792c3333a7911bdfad9e9e9ec58bb48bad93
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-08 04:19:40
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 2: Create PNotification via PBackground r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227687
# ==============================================================================

diff -r 93937caf6d13 -r bb63792c3333 ipc/glue/PBackground.ipdl
--- a/ipc/glue/PBackground.ipdl	Thu Nov 07 17:32:03 2024 +0000
+++ b/ipc/glue/PBackground.ipdl	Thu Nov 07 17:32:03 2024 +0000
@@ -48,6 +48,7 @@
 include protocol PRemoteDecoderManager;
 include protocol PWebTransport;
 include protocol PFetch;
+include protocol PNotification;
 
 include ClientIPCTypes;
 include DOMTypes;
@@ -186,6 +187,21 @@
       Endpoint<PWebTransportParent> aParentEndpoint)
       returns(nsresult rv, uint8_t aReliability); // Actually WebTransportReliabityMode enum
 
+
+  // XXX(krosylight): This should ultimately use nsID instead of scope, see bug 1881812.
+  /**
+   * Finish the setup of a new PNotification top level protocol.
+   */
+  async CreateNotificationParent(
+    Endpoint<PNotificationParent> aParentEndpoint,
+    nsIPrincipal aPrincipal,
+    nsIPrincipal aEffectiveStoragePrincipal,
+    bool aIsSecureContext,
+    nsString aId,
+    nsString aScope,
+    IPCNotificationOptions aOptions
+  ) returns (bool rv);
+
   async PVsync();
 
   async PCameras();
